pkg/controller/daemon/daemon_controller.go

```diff
@@ -1099,7 +1108,8 @@ func (dsc *DaemonSetsController) updateDaemonSetStatus(ds *apps.DaemonSet, hash
 				// Sort the daemon pods by creation time, so that the oldest is first.
 				daemonPods, _ := nodeToDaemonPods[node.Name]
 				sort.Sort(podByCreationTimestamp(daemonPods))
-				pod := daemonPods[0]
+				// Check the newest one; the oldest pods will be deleted later.
+				pod := daemonPods[len(daemonPods)-1]
```

Comment:
Is this a new change or I missed it before? Why is this change needed?

Reply:
A new change: we should check the status according the new pods, the old one will be deleted. Anyway, that's not related to this PR; revert it, and I'll open another PRs for this.